components/eamxx/src/share/field/field.cpp

```diff
@@ -24,20 +24,16 @@ Field::clone() const {
 }
 
 Field
-Field::alias (const std::string& name) const {
-  Field f;
-  f.m_header = get_header().alias(name);
-  f.m_data = m_data;
-  f.m_is_read_only = m_is_read_only;
-  return f;
+Field::clone(const std::string& name) const {
+  return clone(name, get_header().get_identifier().get_grid_name());
 }
 
 Field
-Field::clone(const std::string& name) const {
+Field::clone(const std::string& name, const std::string& grid_name) const {
   // Create new field
   const auto& my_fid = get_header().get_identifier();
   FieldIdentifier fid(name,my_fid.get_layout(),my_fid.get_units(),
-                      my_fid.get_grid_name(),my_fid.data_type());
+                      grid_name,my_fid.data_type());
```

Comment:
I needed to add a grid argument to `Field::clone()` because, in many cases, we are cloning a grid (giving it a new name), constructing a `FM(new_grid)`, but using a field whose ID gives a grid name from the old grid, and the new FM was complaining that the grid was not in the internal grids manager.